beam/core/src/main/java/cz/o2/proxima/beam/direct/io/DirectBatchUnboundedSource.java

```diff
@@ -95,6 +97,40 @@ public static DirectBatchUnboundedSource of(
     public void finalizeCheckpoint() {
       // nop
     }
+
+    @Override
+    public int hashCode() {
+      int hashCode = 73; // because that is the best number
+      return (hashCode
```

Comment:
```
Arrays.hashCode(partitions.stream().map(Partition::getId).sorted().toArray());
```

Reply:
Hm, I don't quite like the idea of copying the array just to compute hash code ... but truth is, that this will not be called too often (maybe might be even dropped, at this was added "to be sure" and not needed anymore).

Reply:
So ... ok :)